GVFS/GVFS.UnitTests.Windows/Windows/Upgrader/ProductUpgraderTests.cs

```diff
@@ -80,6 +80,42 @@ public void UpgradeAvailableOnSlowWhileOnLocalFastRing()
                 expectedUpgradeVersion:UpgradeTests.NewerThanLocalVersion);
         }
 
+        [TestCase]
+        public void RingInNugetFeedURLOverridesUpgradeRing()
+        {
+            Version newVersion;
+            string error;
+            this.SetUpgradeRing(GitHubUpgrader.GitHubUpgraderConfig.RingType.Fast.ToString());
+            this.Upgrader.PretendNewReleaseAvailableAtRemote(UpgradeTests.NewerThanLocalVersion, GitHubUpgrader.GitHubUpgraderConfig.RingType.Fast);
+
+            this.Upgrader.TryQueryNewestVersion(out newVersion, out error).ShouldBeTrue();
+            newVersion.ShouldNotBeNull();
+
+            string feedUrlWithSlowRing = "https://1essharedassets.pkgs.visualstudio.com/_packaging/GVFS@Slow/nuget/v3/index.json";
+            this.LocalConfig.TrySetConfig("upgrade.feedurl", feedUrlWithSlowRing, out error);
+            this.Upgrader.Config.TryLoad(out error).ShouldBeTrue();
+
+            this.Upgrader.TryQueryNewestVersion(out newVersion, out error).ShouldBeTrue();
+            newVersion.ShouldBeNull();
+            error.ShouldContain("Great news");
+        }
+
+        [TestCase]
+        public void NoRingInNugetFeedURLReturnsNullUpgrade()

```

Comment:
Can you add a test where there is no ring set in the nuget feed URL, but there is a ring set in `upgrade.ring`?

Reply:
Added. Please take a look.